feat: Add message truncation in Syslog (#3379)
Logflare/logflare: docs/docs.logflare.com/docs/backends/syslog.mdx

The following values are optional:

- `tls`: (`boolean`) If true, connects via SSL/TLS. Defaults to false (TCP).

> - `max_message_bytes`: (`integer`) Maximum size of the syslog message frame in bytes. If a log exceeds this limit, the JSON body is truncated to fit. If the headers alone exceed this limit, the body is dropped entirely.

Reviewer: should included default value here.

Author: I've added description of the default behavior in https://github.com/Logflare/logflare/pull/3379#discussion_r3123788792

> - `max_message_bytes`: (`integer`) Maximum size of the syslog message frame in bytes. If a log exceeds this limit, the JSON body is truncated to fit. If the headers alone exceed this limit, the body is dropped entirely.

Author: ```suggestion
- `max_message_bytes`: (`integer`) Maximum size of the syslog message frame in bytes. If a log exceeds this limit, the JSON body is truncated to fit. If the headers alone exceed this limit, the body is dropped entirely. Defaults to no limit.
```

> - `max_message_bytes`: (`integer`) Maximum size of the syslog message frame in bytes. If a log exceeds this limit, the JSON body is truncated to fit. If the headers alone exceed this limit, the body is dropped entirely. Defaults to no limit.

Author: ```suggestion
- `max_message_bytes`: (`integer`) Maximum size of the syslog message frame in bytes. If a log exceeds this limit, the JSON body is truncated to fit. If the headers alone exceed this limit, the body is dropped entirely. If not set, the message size is unlimited.
```

> - `max_message_bytes`: (`integer`) Maximum size of the syslog message frame in bytes. If a log exceeds this limit, the JSON body is truncated to fit. If the headers alone exceed this limit, the body is dropped entirely. If not set, the message size is unlimited.

Reviewer: We should set a sensible default for users ( like 75% of conventional max or something like that ) so that this does not need to be explicitly configured

Author: I've set it to 50000 in https://github.com/Logflare/logflare/pull/3379/changes/2c58c21a5249a7d896d5ecce06a44e2837fe6073 (50000/65536 = 0.76).

Also here's what syslog-ng does:

`syslog-ng` defaults `log-msg-size()` to `65536` bytes / 64 KiB, and oversized IETF syslog messages are dropped by default because `trim-large-messages()` defaults to `no`. If trimming is enabled, `syslog-ng` trims to `log-msg-size()` and tags the message.
